### PR TITLE
Ports TG's Laughter Demon

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports TG's laughter demons.

Laughter demons are....well, the same thing as slaughter demons, only they have a penchant for hugging and tickling their victims.

Essentially, mechanics are the same, but the sprite has pink bows on it.

There is one major difference though; on death, the laughter demon will fully heal those who it has consumed--it also doesn't drop a demon heart, but a fluffy little kitten named "laughter".

You can acquire laughter demons, as a wizard, by purchasing them from your spell-book. Each bottle costs 1 point.

:cl: Fox McCloud
add: Adds in laughter demons
add: Can summon laughter demons as a wizard
/:cl:

### CODE REVIEW

File: code/game/gamemodes/miniantags/slaughter/slaughter.dm [hunk 1]
```diff
@@ -181,6 +184,39 @@
...
+	icon_state = "bowmon"
+	icon_living = "bowmon"
+	deathmessage = "fades out, as all of its friends are released from its prison of hugs."
+	loot = list(/mob/living/simple_animal/pet/cat/kitten{name = "Laughter"})
+
+/mob/living/simple_animal/slaughter/laughter/death()
+	for(var/mob/living/M in consumed_mobs)
+		M.revive()
+		if(!M.ckey)
+			for(var/mob/dead/observer/ghost in player_list)
```
Comment (tigercat2000) [inline code/game/gamemodes/miniantags/slaughter/slaughter.dm]: isn't there a proc to get someone's ghost

**Comment by tigercat2000** [comment]
this is hilarious

**Comment by Da-Dman234** [comment]
I'll ask the important question here, can you eat the kitten to gain its powers?

**Comment by Fox-McCloud** [comment]
>I'll ask the important question here, can you eat the kitten to gain its powers?

....no.

**Comment by Spacemanspark** [comment]
What

**Comment by Da-Dman234** [comment]
@Spacemanspark It's best to not ask too many questions

**Comment by Spacemanspark** [comment]
I always ask too many questions.

**Comment by Tastyfish** [comment]
wut

**Comment by Fox-McCloud** [comment]
>wut

Don't you want the tickles?

File: code/game/gamemodes/wizard/spellbook.dm [hunk 2]
```diff
@@ -363,6 +363,21 @@
 	limit = 3
 	category = "Assistance"
 
+/datum/spellbook_entry/item/hugbottle
+	name = "Bottle of Tickles"
+	desc = "A bottle of magically infused fun, the smell of which will \
```
Comment (Aurorablade) [inline code/game/gamemodes/wizard/spellbook.dm]: magically infused fun?

**Comment by Spacemanspark** [comment]
No ERP, Fox.

File: code/game/gamemodes/wizard/spellbook.dm (hunk 2 above)
Comment (TheDZD) [inline code/game/gamemodes/wizard/spellbook.dm]: Don't let @DaveTheHeadcrab find out.

**Pull request merged**